multiqc/modules/biscuit/biscuit.py

```diff
@@ -64,214 +66,210 @@ def __init__(self):
+            # Coverage distribution - q40 cpgs, bottom GC-content
+            ("covdist_q40_cpg_botgc", parse_covdist),
+            # Coverage distribution - q40 cpgs
+            ("covdist_q40_cpg", parse_covdist),
+            # Coverage distribution - q40 cpgs, top GC-content
+            ("covdist_q40_cpg_topgc", parse_covdist),
+            # CpG Retention by Position in Read
+            ("cpg_retention_readpos", parse_retention_readpos),
+            # CpH Retention by Position in Read
+            ("cph_retention_readpos", parse_retention_readpos),
+            # Base-averaged cytosine retention rate
+            ("base_avg_retention_rate", parse_avg_retention),
+            # Read-averaged cytosine retention rate
+            ("read_avg_retention_rate", parse_avg_retention),
         ]
 
-        # Find and parse alignment reports
-        for k in self.mdata:
-            for f in self.find_log_files(f"biscuit/{k}"):
-                s_name = f["fn"]
-                for suffix in file_suffixes:
-                    s_name = s_name.replace(suffix, "")
-                s_name = self.clean_s_name(s_name, f)
-
-                # Add source file to multiqc_sources.txt
-                self.add_data_source(f, s_name=s_name, section=k)
-
-                if s_name in self.mdata[k]:
-                    log.debug(f"Duplicate sample name found in {f['fn']}! Overwriting: {s_name}")
-
-                self.mdata[k][s_name] = getattr(self, f"parse_logs_{k}")(f["f"], f["fn"])
-
-        for k in self.mdata:
-            self.mdata[k] = self.ignore_samples(self.mdata[k])
-
-        n_samples = max([len(self.mdata[k]) for k in self.mdata])
+        n_covdist_samples = 0
+        for file_type, parser_func in file_types:
+            collected_data = {}
+            for f in self.find_log_files(f"biscuit/{file_type}"):
+                parsed_data = parser_func(f["f"], f["fn"])
+                if parsed_data is not None:
+                    s_name = self.clean_s_name(f["s_name"], f)
+
+                    if s_name in collected_data:
+                        log.debug(f"Duplicate sample name found in {f['fn']}! Overwriting: {s_name}")
+
+                    # Add source file to multiqc_sources.txt
+                    self.add_data_source(f, s_name=s_name, section=file_type)
+                    collected_data[s_name] = parsed_data
+
+            collected_data = self.ignore_samples(collected_data)
+            self.biscuit_data[file_type] = collected_data
+
+            # Count the number of coverage distribution samples (denoted by "covdist_*")
+            #
+            # Perform counting here to reduce having to add 12 values together from the biscuit_data
+            # dictionary later
+            #
+            # Short-circuit search at position = 10 (longer than "covdist_", but will ensure that it
+            # is caught if it occurs at the start of the string)
+            if file_type.startswith("covdist_", 0, 10):
```

Comment:
Excessive optimisation..
```suggestion
            if file_type.startswith("covdist_"):
```